### ISSUE TITLE
Incorrect IDE0052 (assigned but never read)

### ISSUE THREAD

**Issue by amcasey** [dotnet/roslyn#30397]
**Version Used**: 2.11.0-beta1-63403-02+32efa2bf484f009a1c8809d782fdaec3b58b937e

**Steps to Reproduce**:

```CSharp
class C
{
    private uint counter; // IDE0052
    public uint Next() => ++counter;
}
```

**Expected Behavior**: No diagnostics

**Actual Behavior**: `IDE0052: Private member 'C.counter' can be removed as the value assigned to it is never read.`

**Comment by amcasey** [comment]
The comment makes it seem like this case was explicitly considered and should not be producing the diagnostic: https://github.com/dotnet/roslyn/blob/6ebb41d4aa7cfcc55d3e1d67914a3de13adc1b00/src/Features/Core/Portable/RemoveUnusedMembers/AbstractRemoveUnusedMembersDiagnosticAnalyzer.cs#L205

**Comment by amcasey** [comment]
Looks like `memberReference.Parent.Parent?.Type == null` is true so the usage is downgraded from `ReadWrite` to `Write`.

**Comment by amcasey** [comment]
FYI @mavasani

**Comment by mavasani** [comment]
Thanks, thought had a test covering this... Will fix it soon.

**Comment by mavasani** [comment]
Likely a conversion is in play here..

**Comment by mavasani** [comment]
I can repro the issue, and the test case is also validating the same scenario. However, it seems `TestMissingInRegularAndScriptAsync` only validates no code actions are offered. I will fix the infrastructure, verify the tests fail, and then fix the analyzer to get the tests passing again.

Thanks @amcasey, your report was helpful!
